server/hotrod/src/main/java/org/infinispan/server/hotrod/AuthenticationHandler.java

```diff
@@ -56,8 +56,8 @@ public AuthenticationHandler(HotRodServer server) {
 
       serverConfig = server.getConfiguration();
       authenticationConfig = server.getConfiguration().authentication();
-      requireAuthentication = authenticationConfig.mechProperties().containsKey(Sasl.POLICY_NOANONYMOUS)
-            && authenticationConfig.mechProperties().get(Sasl.POLICY_NOANONYMOUS).equals("true");
+         requireAuthentication = !authenticationConfig.mechProperties().containsKey(Sasl.POLICY_NOANONYMOUS)
```

Comment:
Do we need to document this in upgrade guide?  Seems this could break some installs.